django_password_validators/password_history/password_validation.py

```diff
@@ -45,15 +56,32 @@ def validate(self, password, user=None):
 
         for user_config in UserPasswordHistoryConfig.objects.filter(user=user):
             password_hash = user_config.make_password_hash(password)
+
             try:
-                PasswordHistory.objects.get(
-                    user_config=user_config,
-                    password=password_hash
-                )
-                raise ValidationError(
-                    _("You can not use a password that is already used in this application."),
-                    code='password_used'
-                )
+                if self.lookup_range == 0:
+                    raise PasswordHistory.DoesNotExist
+
+                current_user_passwords = PasswordHistory.objects.filter(
+                    user_config=user_config).order_by('date')
+
+                if self.lookup_range >= len(current_user_passwords):
+                    PasswordHistory.objects.get(
+                        user_config=user_config,
+                        password=password_hash
+                    )
+                    raise self.validation_error
+
+                else:
+                    # At this point, there are passwords we have to delete
+                    for entry in current_user_passwords[:len(
+                            current_user_passwords) - self.lookup_range]:
+                        entry.delete()
+
+                    if any([entry.user_config == user_config and
```

Comment:
@spokerman12, `any()` will accept a generator expression and hence the square brackets can be dropped, no?